R/roc.R

```diff
@@ -0,0 +1,261 @@
+auc <- function(SE, SP){
+  TPR <- SE
+  FPR <- 1 - SP
+  dFPR <- c(diff(FPR), 0)
+  dTPR <- c(diff(TPR), 0)
+  sum(TPR * dFPR) + sum(dTPR * dFPR)/2
+}
+
+#' @title 
+#' Return a list containing sensitivity (SE) and specificity (SP) values
+#' 
+#' @name 
+#' return_SE_SP
+#'
+#' @description
+#' \code{return_SE_SP} takes in a CFA fit object or vectors of model parameter
+#' estimates and returns a list object containing SE and SP values under partial
+#' and/or strict invariance.
+#'  
+#' @param out list object
+#' 
+#' @return The output will be a list.
+#' @export
+return_SE_SP <- function(cfa_fit,
+                         pmix, 
+                         from = 0.01,
+                         to = 0.25,
+                         by = 0.01,
+                         cutoffs_from = NULL,
+                         cutoffs_to = NULL,
+                         mod_names = c("par", "str"), 
+                         alpha = NULL,
+                         lambda = NULL, 
+                         theta = NULL,
+                         psi = NULL,
+                         nu = NULL, ...) {
+  
+  if(!missing(cfa_fit)) {
+    est <- format_cfa_partinv(cfa_fit, comp = "est")
+    n_g <- cfa_fit@Data@ngroups # number of groups
+    psi <- est$psi
+    lambda <- est$lambda
+    theta <- est$theta
+    alpha <- est$alpha
+    nu <- est$nu
+  }
+  if(missing(cfa_fit) & !is.null(alpha) & !is.null(psi) & 
+     !is.null(lambda) & !is.null(theta)  & !is.null(nu)) {
+    n_g <- length(alpha) # number of groups
+  }
+  
+  propsels <- seq(from = from, to = to, by = by)
+  use <- "propsels"
+  xl <- "Proportion of selection"
```

Comment:
Not used.

Reply:
'propsels' and 'use' are used but xl isn't - thank you for catching it. I commented it out for now, might use it later for the ROC plots